src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -314,20 +321,37 @@ public void Initialize(IConsole console)
                         UpdateWorkingDirectory();
                         await ExecuteInitScriptsAsync();
 
-                        // check if PMC console is actually opened, then only hook to solution load/close events.
+                        // check if PMC console is actually opened, then only hook to solution load events.
                         if (console is IWpfConsole)
                         {
                             // Hook up solution events
                             _solutionManager.Value.SolutionOpened += (_, __) => HandleSolutionOpened();
-                            _solutionManager.Value.SolutionClosed += (o, e) =>
+                        }
+
+                        _solutionManager.Value.SolutionClosed += (o, e) =>
+                        {
+                            var telemetryEvent = new TelemetryEvent(PowerShellExecuteCommand, new Dictionary<string, object>
+                                {
+                                    { NuGetPMCExecuteCommandCount, _pmcExecutedCount},
+                                    { NuGetNonPMCExecuteCommandCount, _nonPmcExecutedCount},
+                                    { NuGetTotalExecuteCommandCount, _pmcExecutedCount + _nonPmcExecutedCount},
+                                    { IsIWpfConsole, console is IWpfConsole}
```

Comment:
Does this mean that when a customer installs a package with install.ps1 and they use PMC in the same session, that we generate two telemetry events? DOes NuGet create two powershell hosts?

Reply:
Yes, PMC and PMUI creates 2 separate powershell hosts. So it'll be 2 events, and maybe we can make it single event but it involves static variables which I prefer not to do it.

Reply:
That's very interesting. I didn't know that.

An alternative to using statics, is to raise events in the PowerShell host, and have something running in the "higher level" Visual Studio code listen to the events and aggregate the data. My package source telemetry was problematic with regards to data quality and being able to answer questions, but I think it's the first NuGet telemetry that is aggregated, rather than "emit at source". I used a .NET event, but now I think a ETW event would be better. Listening to ETW events is a bit more difficult than subscribing to .NET events, but it eliminates the need for public APIs (public static event).

Basically, it could work the similarly to System.Net.Http's diagnostics: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/HttpDiagnosticsGuide.md

Reply:
What is the relationship between .Net event and EWT events? Is EWT events does produce .Net event?
I looked that `System.Net.Http's diagnostics` but code looks snippet of code, rather than full code not sure if I can implement it this time.
Maybe static variable is easier to understand this one.

Reply:
> What is the relationship between .Net event and EWT events?

There is none. One is internal to the .NET process, the other is designed to be used for high performance monitoring of applications running on windows, with the side-effect that it's a convenient way to pass messages between components in a decoupled way.

Before making changes, I think it will be valuable to have a discussion about the telemetry we want to collect, and how you want to query it.  If you have one telemetry event with two measurements, the complexity in querying it different compared to having two events, each with a single measurement, but you have to filter which of the events are relevant to the query.

Reply:
I combined them into 1 event emit now it would be much easier, please check .
We already held discussion and end conclusion was up to me to decide it.